sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -221,14 +221,14 @@ class BooleanSimplificationSuite extends PlanTest with ExpressionEvalHelper with
 
   test("Complementation Laws - null handling") {
     checkCondition('e && !'e,
-      testRelationWithData.where(If('e.isNull, Literal.create(null, BooleanType), false)).analyze)
+      testRelationWithData.where(And(Literal(null, BooleanType), 'e.isNull)).analyze)
```

Comment:
Although I think it is better to test `BooleanSimplification` like original, but `BooleanSimplificationSuite`'s optimizer mixes `SimplifyConditionals` and `BooleanSimplification`.

Reply:
Yeah. What is the recommended way for adding test suite for optimizers? should each test suite only use the corresponding optimizer rule, or it's OK to also add those relevant ones?

Reply:
Usually we only test corresponding rule in each test suite. But some suites use multiple rules like this one. For this one, although we can still only test `BooleanSimplification` if we add another optimizer, but seems too much. Maybe currently it's good enough. Maybe add a comment.